lib/types/cql-utils.js

```diff
@@ -137,6 +137,35 @@ function encodeTuple(value, type) {
     return [rust.ComplexType.remapTupleSupportType(newTypes), res];
 }
 
+/**
+ * Wraps UDT into format recognized by ParameterWrapper
+ * @param {Object} value
+ * @param {rust.ComplexType} type
+ * @returns {Array<rust.ComplexType|Array<Array<String, any>>>} Returns tuple: [rust.ComplexType, Array<any>]
+ */
+function encodeUdt(value, type) {
+    let names = type.getFieldNames();
+    let types = type.getInnerTypes();
+    let newTypes = [];
+    let res = [];
+
+    if (names.length !== types.length) {
+        throw new TypeError(
+            `Expected ${types.length} field names in UDT, obtained ${names.length}`,
+        );
+    }
+
+    for (let i = 0; i < names.length; i++) {
+        const element = getWrapped(
+            types[i],
+            value[names[i]] !== undefined ? value[names[i]] : null,
+        );
+        newTypes.push(element[0]);
+        res.push([names[i], element[1]]);
+    }
```

Comment:
:wrench: Please explain (in a comment) the logic around `undefined`/`null` here.

Reply:
What are the possible cases? What is their semantics?